sql/dd/impl/upgrade/dd.cc

```diff
@@ -1031,6 +1032,124 @@ bool update_object_ids(THD *thd, const std::set<String_type> &create_set,
 }
 /* purecov: end */
 
+/*
+  Update myrocks internal data during dd_engine change from innodb to rocksdb
+
+  @param  thd                      Thread context.
+  @param  old_schema_name          Name of the old(from) schema
+  @param  new_schema_name          Name of the new(to) schema
+  @param  table_names              table name set to be updated
+
+  @returns false if success, true otherwise.
+*/
+bool update_myrocks_table_name(THD *thd, const String_type &old_schema_name,
+                               const String_type &new_schema_name,
+                               const std::set<String_type> &table_names) {
+  assert(default_dd_storage_engine == DEFAULT_DD_ROCKSDB);
+  assert(bootstrap::DD_bootstrap_ctx::instance().is_dd_engine_change());
+
+  auto *rocksdb_ddse = ha_resolve_by_legacy_type(thd, DB_TYPE_ROCKSDB);
+  if (!rocksdb_ddse) {
+    assert(false);
+    return true;
+  }
+  auto *handler = get_new_handler(nullptr, false, thd->mem_root, rocksdb_ddse);
+  if (!handler) {
+    assert(false);
+    return true;
+  }
+  int err = false;
+  for (auto table_name : table_names) {
```

Comment:
The lack of reference creates a new value every time
```suggestion
  for (const auto &table_name : table_names) {
```